core/src/avm2/object.rs

```diff
@@ -1076,6 +1079,14 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
         None
     }
 
+    /// Unwrap this object's `QName`, if the object is a boxed QName.
+    fn as_qname(&self) -> Option<Ref<QName<'gc>>> {
```

Comment:
can we use a standard `as_qname_object()` instead?
This way we could only have `init_qname` on `QNameObject` only, instead of trait-wide.